### PR TITLE
Added information about using Castle.Windsor

### PR DESCRIPTION
**Pull request by schwarzie2478**
as an dependency injection container inside ASP.Net Core
 - Added reference in ReadMe.md
 - Added link to nuget package of Castle.Facilities.AspNetCore
 - Added sample code for usage of the container

### CODE REVIEW

**Comment by schwarzie2478** [comment]
I could use some help getting this unit test to run, I was not succesful in getting my system configured succesfully to run the unit tests.  The build is succusful, I'm at a loss get the test running for Windsor.

File: src/DependencyInjection/DI.External.Tests/test/SkippableDependencyInjectionSpecificationTests.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ public abstract class SkippableDependencyInjectionSpecificationTests: Dependency
 
         protected sealed override IServiceProvider CreateServiceProvider(IServiceCollection serviceCollection)
         {
-            foreach (var stackFrame in new StackTrace(1).GetFrames().Take(2))
+            foreach (var stackFrame in new System.Diagnostics.StackTrace(1).GetFrames().Take(2))
```
Comment (dougbu) [inline src/DependencyInjection/DI.External.Tests/test/SkippableDependencyInjectionSpecificationTests.cs]: Why is this change necessary?

File: src/DependencyInjection/DI.External.Tests/test/Windsor.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+using System.Text;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public class WindsorDependencyInjectionSpecificationTests : SkippableDependencyInjectionSpecificationTests
+    {
+        private static readonly WindsorContainer Container = new WindsorContainer();
+        private class HomeController
+        { }
+        public override string[] SkippedTests => throw new NotImplementedException();
```
Comment (anurse) [inline src/DependencyInjection/DI.External.Tests/test/Windsor.cs]: Throwing here is causing all the tests to fail. If you have no skipped tests you need to return an empty array:

```suggestion
        public override string[] SkippedTests => Array.Empty<string>();
```

File: src/DependencyInjection/DI.External.Tests/test/SkippableDependencyInjectionSpecificationTests.cs (hunk 1 above)
Comment (schwarzie2478) [reply src/DependencyInjection/DI.External.Tests/test/SkippableDependencyInjectionSpecificationTests.cs]: This namespace shouldn't be needed but I had the strange behavior that I couldn't get to resolve StackTrace unless I fully specified it. Using statement kept being ignored.  Both intellisense and the compiler complained about it.  I had the option enable to let VS use preview tooling and the latest preview4 build was just installed.

Comment (pakrym) [reply src/DependencyInjection/DI.External.Tests/test/SkippableDependencyInjectionSpecificationTests.cs]: This shouldn't be required.

**Comment by pakrym** [comment]
As long as the test passes this seems to be a fine change, but it's not passing ATM.

**Comment by davidfowl** [comment]
The tests all fail with 

```
System.ArgumentException : Facility of type 'Castle.Facilities.AspNetCore.AspNetCoreFacility' has already been registered with the container. Only one facility of a given type can exist in the container.
```

PS: Castle should drop the ASP.NET Core branding from this plugin.

File: src/DependencyInjection/DI.External.Tests/test/SkippableDependencyInjectionSpecificationTests.cs (hunk 1 above)
Comment (schwarzie2478) [reply src/DependencyInjection/DI.External.Tests/test/SkippableDependencyInjectionSpecificationTests.cs]: Removed it from the code.  Intellisence still complains but the code builds and the tests run ok.

File: src/DependencyInjection/DI.External.Tests/test/Windsor.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
+using Castle.Facilities.AspNetCore;
```
Comment (Eilon) [inline src/DependencyInjection/DI.External.Tests/test/Windsor.cs]: Missing license header in this file.

File: eng/Dependencies.props [hunk 4]
```diff
@@ -83,6 +83,7 @@ and are generated based on the last package release.
     <LatestPackageReference Include="Stashbox.Extensions.Dependencyinjection" Version="2.6.3" />
     <LatestPackageReference Include="StructureMap.Microsoft.DependencyInjection" Version="1.4.0" />
     <LatestPackageReference Include="Unity.Microsoft.DependencyInjection" Version="2.1.3" />
+    <LatestPackageReference Include="Castle.Facilities.AspNetCore" Version="5.0.0" />
```
Comment (davidfowl) [inline eng/Dependencies.props]: This repository shouldn't depend on ASP.NET Core. Can you remove that from the container implementation? None of the other containers rely on ASP.NET Core here.

Comment (schwarzie2478) [reply eng/Dependencies.props]: I'm not the maintainer of this package, but I opened an discussion on that repo page. [https://github.com/castleproject/Windsor/issues/487](url)

I'm interested to hear from them too...

File: src/DependencyInjection/DI.External.Tests/test/SkippableDependencyInjectionSpecificationTests.cs (hunk 1 above)
Comment (ericstj) [reply src/DependencyInjection/DI.External.Tests/test/SkippableDependencyInjectionSpecificationTests.cs:14]: It's possible Intellisense was finding an internal StackTrace member in some other namespace...  Still intellisense and compile ought to be using the same algorithm.  If you can reproduce it we can file a bug against roslyn.

**Comment by ericstj** [comment]
I believe this code has now moved to dotnet/runtime /cc @maryamariyan 

This PR will need to be opened there.  We should also avoid an upstack dependency on ASPNetCore as @davidfowl suggests if we add this to dotnet/runtime tests.

**Comment by maryamariyan** [comment]
Yes DI.External tests are moved to dotnet/runtime repo:

https://github.com/dotnet/runtime/tree/master/src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests
